app/models/user.rb

```diff
@@ -87,4 +88,17 @@ def send_devise_notification(notification, *args)
   def decorate
     UserDecorator.new(self)
   end
+
+  private
+
+  # used by zxcvbn
+  def weak_words
+    [APP_NAME]
+  end
+
+  # method required by zxcvbn
+  def password_required?
+    confirmable = confirmed? || confirmation_token.present?
+    password.present? && confirmable
+  end
```

Comment:
Not sure how to apply a Devise plugin to something other than the User model, w/o a lot of gymnastics. I'd prefer to keep validation as close to the model as possible anyway, since I do not trust that a form is our only data input channel.